isisd/isis_adjacency.c

```diff
@@ -328,15 +328,14 @@ void isis_adj_state_change(struct isis_adjacency **padj,
 				circuit->upadjcount[level - 1]--;
 				if (circuit->upadjcount[level - 1] == 0)
 					isis_tx_queue_clean(circuit->tx_queue);
-
-				hook_call(isis_adj_state_change_hook, adj);
 				del = true;
 			}
 		}
 	}
 
 	if (del) {
 		isis_delete_adj(adj);
+		hook_call(isis_adj_state_change_hook, adj);
```

Comment:
`adj` is a dangling pointer after the previous line.

What's the motivation behind this change?

Reply:
The motivation is that this hook triggers in the very end the SPF re-calculation. This needs to happen after `isis_delete_adj` since only then the correct SPF data is set and those calculations will be correct.

However, what I did here is bogus of course in terms of memory management :-/. I will find a better solution. This is also something the CI doesn't like.